api/metrics/src/main/java/io/opentelemetry/api/metrics/LongUpDownCounterBuilder.java

```diff
@@ -5,14 +5,42 @@
 
 package io.opentelemetry.api.metrics;
 
+import java.util.function.Consumer;
+
 /** Builder class for {@link LongUpDownCounter}. */
-public interface LongUpDownCounterBuilder extends SynchronousInstrumentBuilder {
-  @Override
+public interface LongUpDownCounterBuilder {
+  /**
+   * Sets the description for this instrument.
+   *
+   * <p>Description strings should follow the instrument description rules:
+   * https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-description
+   */
   LongUpDownCounterBuilder setDescription(String description);
 
-  @Override
+  /**
+   * Sets the unit of measure for this instrument.
+   *
+   * <p>Unit strings should follow the instrument unit rules:
+   * https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-unit
+   */
   LongUpDownCounterBuilder setUnit(String unit);
 
-  @Override
+  /** Sets the counter for recording {@code double} values. */
+  DoubleUpDownCounterBuilder ofDoubles();
+
+  /**
+   * Builds and returns a {@code LongUpDownCounter} with the desired options.
+   *
+   * @return a {@code LongUpDownCounter} with the desired options.
+   */
   LongUpDownCounter build();
+
+  /**
+   * Builds this asynchronous insturment with the given callback.
+   *
+   * <p>The callback will only be called when the {@link Meter} is being observed.
+   *
+   * @param callback A state-capturing callback used to observe values on-demand.
+   */
+  void buildWithCallback(Consumer<ObservableLongMeasurement> callback);
```

Comment:
I'd like to see "observable" somehow incorporated into the naming convention of methods that access async instruments. Its recommended in the spec and I think it better clarifies the async distinction than "callback". 

Maybe `buildObservable`?

Reply:
@jkwatson @anuraaga for comment.   This was a specific deviation to fit the java "builder" convention, so I defer to Java SiG here.

Reply:
The spec calls this an "observable" I think? In that case, probably migrating to that verbiage would be helpful. I don't want to hold up this PR for that change, though.